### PR TITLE
Feature/add scm filters for branch and pr separately

### PR DESCRIPTION
**Pull request by ThejashwiniSwamy**
Handle filtering pull requests separately from branches.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by nrayapati** [comment]
Thank you for making this PR @ThejashwiniSwamy.

Really appreciate if someone can help review this change and have it available as a release. Thank you very much in advance!

File: src/main/java/net/gleske/scmfilter/impl/trait/RegexSCMBranchFilterTrait.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                        || head instanceof ChangeRequestSCMHead2)) {
+                    return !getBranchPattern().matcher(head.getName()).matches();
+                }
+
+                return false;
+            }
+        });
+    }
+
+    @Symbol("RegexSCMBranchFilter")
```
Comment (jonesbusy) [inline src/main/java/net/gleske/scmfilter/impl/trait/RegexSCMBranchFilterTrait.java:120]: Better to start symbol with lower case

File: src/main/java/net/gleske/scmfilter/impl/trait/RegexSCMPRDestinationFilterTrait.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+                return false;
+            }
+        });
+    }
+
+    /**
+     * Our descriptor.
+     */
+    @Symbol("RegexSCMPRDestinationFilter")
```
Comment (jonesbusy) [inline src/main/java/net/gleske/scmfilter/impl/trait/RegexSCMPRDestinationFilterTrait.java:121]: lower case symbol

File: src/main/java/net/gleske/scmfilter/impl/trait/RegexSCMPROriginFilterTrait.java [hunk 3]
```diff
@@ -161,28 +104,22 @@ private Pattern getTagPattern() {
...
+                return false;
             }
         });
     }
 
     /**
      * Our descriptor.
      */
-    @Symbol("headRegexFilterWithPRFromOrigin")
+    @Symbol("RegexSCMPROriginFilter")
```
Comment (jonesbusy) [inline src/main/java/net/gleske/scmfilter/impl/trait/RegexSCMPROriginFilterTrait.java:122]: This is a breaking change

**Review by jonesbusy — CHANGES_REQUESTED** [review]

**Comment by jonesbusy** [comment]
Hi,

I took over minimal maintenance of this plugin.

This PR introduce breaking changes.

I'm totally fine to introduce new traits to separate branch and PR filtering. 

But we should not break compatibility with user already using those traits. 

Also adding tests would be appreciated.

Best regards,

**Comment by jonesbusy** [comment]
Also please rebase your PR on main

**Comment by samrocketman** [comment]
This is a major refactor of existing functionality.  You could add a checkbox so that the filter applies only to pull requests and not branches.

There's no need to have a separate filter just for branches.  The existing SCM branch filter already covers filtering by branch.  This plugin provides a gap that the branch scm filter did not meet (PRs destined for branch).

If you want this to optionally apply to only PRs then the right approach is to introduce a new checkbox to the UI which applies only to pull requests and not branches.

This refactor would duplicate existing core functionality with exception for the purpose this plugin provides (PRs based on source or destination branch).

**Comment by samrocketman** [comment]
Closing since it would duplicate the built in branch filters provided by core pipeline.

If you think I misunderstood the feature feel free to reply to my comments.

**Comment by nrayapati** [comment]
@samrocketman, @jonesbusy  we're trying to achieve a similar functionality to the author of this closed PR. We'd like to organize our builds into separate folders based on the branch type:

* Main Folder: Contains the `main` build and PR builds for the `main` branch.
* Releases Folder: Contains builds for all `release-*` branches and their associated PR builds.
* Dev Folder: Contains all remaining builds.

Could you please guide us on how to configure the current version of this plugin to support this use case without including the changes from the closed PR? We'd appreciate any options or settings we can leverage to implement this folder structure without modifying the plugin itself.

Thank you for maintaining this awesome plugin!

**Pull request closed without merging**